nilearn/decomposition/tests/test_multi_pca.py

```diff
@@ -16,54 +16,75 @@ def _tmp_dir():
     return tempfile.tempdir + os.sep
 
 
-def test_multi_pca():
-    # Smoke test the _MultiPCA
-    # XXX: this is mostly a smoke test
+def _make_multi_pca_test_data(with_activation=True):
+    """Create a multi-subject dataset with or without activation."""
     shape = (6, 8, 10, 5)
     affine = np.eye(4)
     rng = np.random.RandomState(0)
 
-    # Create a "multi-subject" dataset
     data = []
     for _ in range(8):
         this_data = rng.normal(size=shape)
-        # Create fake activation to get non empty mask
-        this_data[2:4, 2:4, 2:4, :] += 10
+        if with_activation:
+            this_data[2:4, 2:4, 2:4, :] += 10
         data.append(nibabel.Nifti1Image(this_data, affine))
 
     mask_img = nibabel.Nifti1Image(np.ones(shape[:3], dtype=np.int8), affine)
+
+    return data, mask_img, shape, affine
+
+
+def test_multi_pca():
+    """Test that the components are the same if we put twice the same data, \
+    and that fit output is deterministic"""
+    data, mask_img, _, _ = _make_multi_pca_test_data()
+
     multi_pca = _MultiPCA(mask=mask_img, n_components=3, random_state=0)
-    # fit to the data and test for masker attributes
```

Comment:
Maybe it's bad taste, but I find these comments helpful.

Reply:
actually this should probably be extracted into another test and turn the comment into the name of the function